controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -391,6 +399,11 @@ protected void shutDown() throws Exception {
         log.info("Initiating controller service shutDown");
 
         try {
+            if (healthServiceManager != null) {
+                log.info("Stopping the HealthService.");
+                healthServiceManager.close();
```

Comment:
For some reason, these method are called once in `shutDown` and once in `close. I see that the other components are closed again in `close`. I think it would be worth it to add it there too.